### PR TITLE
Extensions Sharing

### PR DESCRIPTION
**Pull request by trooper2013**
Changes related to extension enablement. Replaced [NSUserDefaults standardUserDefaults] with
[NSUserDefaults msdkUserDefaults]. Incorporated @khawkins  feedback from PR #1709

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1527,6 +1540,7 @@
 			isa = PBXHeadersBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				B76ACC521D5BD75E004BF7DB /* NSUserDefaults+SFAdditions.h in Headers */,
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj]: Make sure this header is a part of both targets of `SalesforceSDKCore` - the dynamic framework target and the static target.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSUserDefaults+SFAdditions.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/*
+ Copyright (c) 2015, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSUserDefaults+SFAdditions.h]: `2016`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSUserDefaults+SFAdditions.m [hunk 3]
```diff
@@ -0,0 +1,39 @@
+/*
+ Copyright (c) 2015, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSUserDefaults+SFAdditions.m]: `2016`.

**Comment by bhariharan** [comment]
Looks pretty good otherwise. We might need some upgrade/migration steps for existing apps.

**Comment by bhariharan** [comment]
Oh, and one more thing - you'll need to add entries for this new file in `SalesforceSDKCore.podspec`, but we can do that later. I think I might have to do that for `AILTN` too, so I'll take care of that.

**Comment by khawkins** [comment]
👍  We can figure out upgrade steps in the next round.

**Comment by bhariharan** [comment]
👍

**Pull request merged**